src/utils/evm/parsing.ts

```diff
@@ -0,0 +1,44 @@
+import { ethers } from 'ethers'
+import { parseMethod } from '../parsing'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function extractContractAddress(rawData: Record<string, any>) {
+  const [firstParameter] = rawData.params
+  const method = parseMethod(rawData)
+
+  if (method === 'eth_sendRawTransaction') {
+    // firstParameter is the raw tx hex
+    return decodeEthRawTxAddress(firstParameter)
+  } else if (method === 'eth_call') {
+    return firstParameter?.to
+  } else if (method === 'eth_getLogs') {
+    return firstParameter?.address
+  } else if (
+    method === 'eth_getCode' ||
```

Comment:
add this to an array and check with `includes`

Reply:
Why add it to an array? Feels like an extra step that wastes CPU cycles.

Reply:
code readability

Reply:
I do, not my PR so the owner can do it if they want, just stated my opinion

Reply:
> I do, not my PR so the owner can do it if they want, just stated my opinion

I moved this to a switch, more readable there :)